core/app/models/spree/order.rb

```diff
@@ -438,14 +438,20 @@ def ensure_line_items_are_in_stock
     end
 
     def empty!
-      line_items.destroy_all
-      updater.update_item_count
-      adjustments.destroy_all
-      shipments.destroy_all
-      state_changes.destroy_all
+      if completed?
+        raise Spree.t(:cannot_be_emptied_order_processed_already)
```

Comment:
@tanmay3011 Yes, I meant translation key.
```raise Spree.t(:cannot_empty_completed_order)```
instead of
```raise Spree.t(:cannot_be_emptied_order_processed_already)```.